Fabric.Authorization.API/Services/IdentitySearchService.cs

```diff
@@ -101,13 +119,13 @@ public class IdentitySearchService
                 });
             }
 
-            var userDetails =
+            var fabricIdentityUserResponse =
                 await _identityServiceProvider.Search(request.ClientId, userList.Select(u => $"{u.SubjectId}:{u.IdentityProvider}"));
 
-            if (userDetails != null)
+            if (fabricIdentityUserResponse != null && fabricIdentityUserResponse.HttpStatusCode == System.Net.HttpStatusCode.OK)
             {
                 // update user details with Fabric.Identity response
-                foreach (var user in userDetails)
+                foreach (var user in fabricIdentityUserResponse.Results)
                 {
                     var userSearchResponse = userList.FirstOrDefault(u => u.SubjectId == user.SubjectId);
```

Comment:
should identity provider also be used here?

Reply:
@kpaul, not sure I follow?

Reply:
when searching for users returned from identity, will it be enough to only match on subject id or should we also match on identity provider